b2/sync/folder.py

```diff
@@ -113,6 +114,20 @@ def ensure_non_empty(self):
         if not os.listdir(self.root):
             raise Exception('Directory %s is empty' % (self.root,))
 
+    def _is_file_readable(self, local_path, reporter):
```

Comment:
I suggest `reporter=None` as the method is clearly expecting this, this way we can let the user of this method do `self._is_file_readable(local_path)` instead of `self._is_file_readable(local_path, reporter=None)` which is a bit tedious or `self._is_file_readable(local_path, None)` which looks weird.

I am also thinking if `_is_file_readable` should be located in `sync.folder.AbstractFolder()` class? It strikes me that a subclass is unlikely to shadow it and the method is not referencing `self` at any point. I think a more appropriate place could be a function in `b2.utils`. Another argument is that for testing this method we don't really need an instance of AbstractFolder subclass (that kind of test would look rather awkward, we'd have to write a class that inherits from AbstractFolder... to work around that we could make `_is_file_readable` a classmethod, but that's still clearly not perfect, I think b2.utils is the way to go). If you agree, please move it, alternatively lets discuss it.